Clinical-Genomics/scout
# Matching causatives found in research variants from other cases

Fix #3512 

Current code performs a strict matching of variants _ids (created as md5 keys using chr, start, ref, alt, **clinical or research**), so when a case has clinical variants there won't be any matches for those variants in another case that has research variants marked as causative. 

I've changed the code to look for causatives also among research variants by using the 2 possible display names of a variant:
- chr_start_ref_alt_clinical
- chr_start_ref_alt_research

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the paxa software installed on `hasta`:
 - Log in into hasta: `ssh <USER.NAME>@hasta.scilifelab.se`
 - Activate the staging environment with the command `us` (Use Stage)
 - Run the command `paxa` and follow the instructions. Make sure to specify scout-stage as the resource you request and the server cg-vm1 as server.
1. Log out from the hasta server.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, log out from `cg-vm1` and log in again in the `hasta` server, repeat the `hasta` and `paxa` procedure, which will release the allocated resource (scout-stage) to be used for testing by other users.
</details>


**How to test, locally**:
1. Load the the demo case 643595: `scout --demo load case scout/demo/643595.config.yaml`
2. Load research variants for the demo case 643594: `scout --demo load research -c internal_id -i cust000 --force`
3. Go to case 643594, **research variants**, and mark as causative the usual POT1 var: http://localhost:5000/cust000/643594/80e1bdac78e83339117e4929f116a147
4. On main branch, if you go to the corresponding var of the close case 643595, it won't show any matching causatives
5. Switch to this branch and check that now the matching causative (research var on case 643594) is shown

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
